ansible/roles/debops.ansible_plugins/filter_plugins/debops_filter_plugins.py

```diff
@@ -24,44 +25,75 @@
 from past.builtins import basestring
 from operator import itemgetter
 
-try:
-    unicode = unicode
-except NameError:
-    # py3
-    unicode = str
-
 __metaclass__ = type
 
 
-def _parse_kv_value(current_data, new_data, data_index, *args, **kwargs):
+def _check_if_key_in_nested_dict(key, dictionary):
+    if isinstance(dictionary, dict):
+        for k, v in dictionary.items():
+            if k == key:
+                return True
+            elif isinstance(v, dict):
+                if _check_if_key_in_nested_dict(key, v):
+                    return True
+            elif isinstance(v, list):
+                for d in v:
+                    if _check_if_key_in_nested_dict(key, d):
+                        return True
+
+    return False
+
+
+def _handle_copy_id_from(parsed_config, element, current_param):
+    if 'copy_id_from' in element:
+        if element['copy_id_from'] in parsed_config:
+            id_src = element['copy_id_from']
+            current_param['id'] = (
+                int(parsed_config[id_src]['id']) +
+                int(parsed_config[id_src].get('weight', 0)))
+
+
+def _handle_weight(element, current_param):
+    if 'weight' in element:
+        current_param['weight'] = (
+            int(element.get('weight',
+                current_param.get('weight', 0))) +
+            int(current_param.get('weight', 0)))
+
+
+def _get_real_weight(current_param):
+    return int(current_param['id']) + int(current_param['weight'])
+
+
+def _parse_kv_value(current_data, new_data, data_index):
     """Parse the parameter values and merge
     with existing ones conditionally.
     """
 
     if 'value' in new_data:
-        old_value = current_data.get('value', None)
+        old_value = current_data.get('value')
         old_state = current_data.get('state', 'present')
-        new_value = new_data.get('value')
+        new_value = new_data['value']
 
         if isinstance(new_value, (basestring, int, float, bool)):
             if (old_value is None or isinstance(old_value,
-                                                (str, unicode, int,
+                                                (basestring, int,
                                                  float, bool, dict))):
-                current_data.update({'value': new_value})
+                current_data['value'] = new_value
 
-            if (old_value is not None and old_state in ['comment'] and
-                    current_data['state'] != 'comment'):
-                current_data.update({'state': 'present'})
+            # TODO(drybjed): This never evaluates to true.
+            #  if (old_value is not None and old_state in ['comment'] and
+            #          current_data['state'] != 'comment'):
+            #      current_data['state'] = 'present'
```

Comment:
I need to write more comments in my code. :-) This IIRC was meant to activate a given configuration entry that is commented out in role defaults if somebody defined an entry with the same `name` and some value, basically switching the state from `comment` to `present` without explicitly defining the `state` parameter. I suspect that this might be too much magic, so perhaps this should be removed altogether. Let's leave it commented out now to see if anything breaks.

Reply:
Difference is that if an entry has an `init` state, it would not show up in the generated config file at all unless it was redefined in the Ansible inventory. With the `comment` state it will show up, but commented out, unless it is redefined in the inventory. The difference is mostly cosmetic I suppose, for example if you want to keep some of the commented out parameters in the Postfix `main.cf` configuration file unless they are changed by the user.

Reply:
Thanks. I only used `init` for now.